security/pfSense-pkg-acme/files/usr/local/www/acme/acme_accountkeys_edit.php

```diff
@@ -0,0 +1,503 @@
+		$domain_name    = $server['name'];
+		if (!is_hostname($domain_name)) {
+			$input_errors[] = "The field 'Domainname' does not contain a valid hostname.";
+		}
+	}
+	$a_actions = $actionslist->acme_htmllist_get_values();
+
+	$accountkey = array();
+	if(isset($id) && $a_accountkeys[$id]) {
+		$accountkey = $a_accountkeys[$id];
+	}
+		
+//	echo "newname id:$id";
+	if (!empty($accountkey['name']) && ($accountkey['name'] != $_POST['name'])) {
+		//old $accountkey['name'] can be empty if a new or cloned item is saved, nothing should be renamed then
+		// name changed:
+		$oldvalue = $accountkey['name'];
+		$newvalue = $_POST['name'];
+		
+		$a_accountkeys = &$config['installedpackages']['acme']['accountkeys']['item'];
+		if (!is_array($a_accountkeys)) {
+			$a_accountkeys = array();
+		}
+	}
+
+	if($accountkey['name'] != "") {
+		$changedesc .= " modified pool: '{$accountkey['name']}'";
+	}
+	$accountkey['a_domainlist']['item'] = $a_domains;
+	$accountkey['a_actionlist']['item'] = $a_actions;
+
+	$accountkey['accountkey'] = base64_encode($_POST['accountkey']);
+	global $simplefields;
+	foreach($simplefields as $stat) {
+		update_if_changed($stat, $accountkey[$stat], $_POST[$stat]);
+	}
+	
+	if (isset($id) && $a_accountkeys[$id]) {
+		$a_accountkeys[$id] = $accountkey;
+	} else {
+		$a_accountkeys[] = $accountkey;
+	}
+	if (!isset($input_errors)) {
+		if ($changecount > 0) {
+			touch($d_acmeconfdirty_path);
+			write_config($changedesc);
+		}
+		echo "<pre/>";
+		//print_r($config['installedpackages']['acme']);
+		header("Location: acme_accountkeys.php");
+		exit;
+	}
+}
+
+//$closehead = false;
+$pgtitle = array("Services", "Acme", "Certificate options: Edit");
+include("head.inc");
+display_top_tabs_active($acme_tab_array['acme'], "accountkeys");
+
+// 'processing' done, make all simple fields usable in html.
+foreach($simplefields as $field){
+	$pconfig[$field] = htmlspecialchars($pconfig[$field]);
+}
+
+?>
+<!--/head-->
+<?php
+if (isset($input_errors)) {
+	print_input_errors($input_errors);
+}
+
+$counter=0;
+
+$form = new \Form;
+
+$section = new \Form_Section('Edit Certificate options');
+$section->addInput(new \Form_Input('name', 'Name', 'text', $pconfig['name']
+))->setHelp('');
+$section->addInput(new \Form_Input('desc', 'Description', 'text', $pconfig['desc']));
+
+$section->addInput(new \Form_Select(
+	'acmeserver',
+	'Acme Server',
+	$pconfig['acmeserver'],
+	form_keyvalue_array($a_acmeserver)
+));
+
+$section->addInput(new \Form_Textarea(
+	'accountkey',
+	'Account key',
+	$pconfig['accountkey']
+))->setNoWrap();
+
+$section->addInput(new \Form_StaticText(
+	'', 
+	"<a id='btncreatekey' class='btn btn-sm btn-primary'>"
+		. "<i id='btncreatekeyicon' class='fa fa-check'></i> Create new account key</a>"
+));
+
+$section->addInput(new \Form_StaticText(
+	'Acme account registration',
+	"<a id='btnregisterkey' class='btn btn-sm btn-primary'>"
+		. "<i id='btnregisterkeyicon' class='fa fa-check'></i> Register acme account key</a>"
```

Comment:
Register button should not use a checkmark. Use an icon that implies account registration, like fa-user, fa-key, maybe fa-lock.